queue/job.go

```diff
@@ -43,8 +43,13 @@ type Acknowledger interface {
 // NewJob creates a new Job with default values, a new unique ID and current
 // timestamp.
 func NewJob() *Job {
+	u, err := uuid.NewV4()
+	if err != nil {
```

Comment:
How can this fails?

Reply:
maybe we should use uuid.Must()

Reply:
It reads from random https://github.com/satori/go.uuid/blob/c596ec57260fd2ad47b2ae6809d6890a2f99c3b2/generator.go#L168

Reply:
Must panics if there is an error but I'm not fan of panic in libraries. The code using a method that gives an error should be the one deciding if it is a fatal error or it can continue. Maybe retrying or failing that job.

Reply:
the problem here is that we are changing the expected behavior, now that constructor can return nil. Do we have this case checked in the code where this method is called? if not, we are gonna have a nil pointer panic.

Reply:
If there is no panic, there should be an error returned. Which, of course, changes all the API of the function and we have more breaking changes to deal with. But at least is explicit. Returning nil may introduce silent bugs.

Reply:
So the best approach is adding a panic now and opening an issue to change the NewJob and whichever is using the library to receive errors?

Reply:
IMO, the best option is to return an error, even if it implies more changes in other libraries. I mean, it's not like many other projects use framework. Possibly only core-retrieval uses it

Reply:
In borges this is used in these files (plus tests):

* `executor.go`
* `producer.go`

I'm not sure this is used in other projects. Maybe it's the right time to change the API.

Reply:
I'm changing the code to return error and will change borges to be able to use this new version.